UnitTests/GTMSessionFetcherFetchingTest.m

```diff
@@ -2372,24 +2373,165 @@ - (nonnull XCTestExpectation *)expectationForStartFetchThatStopsFetching:
+//                               extra work/configuration.
+// @param postBegin              An optional block to run right after the `-beginFetch...` call.
+//
+- (void)runStopFetchingNoCallbackTestWithNotifications:(BOOL)triggersNotifications
+                                    bypassServiceCheck:(BOOL)bypassServiceCheck
+                                              preBegin:
+                                                  (nullable StopFetchingCallbackTestBlock)preBegin
+                                             postBegin:
+                                                 (nullable StopFetchingCallbackTestBlock)postBegin {
+  if (!_isServerRunning) return;
+
+  // Sanity check the usage of this helper.
+  XCTAssertTrue(preBegin != nil || postBegin != nil,
+                @"preBegin and postBegin can't both be nil, you must provide at least one, and one "
+                @"of them has to call `-stopFetching`");
+  XCTAssertTrue(triggersNotifications || (_fetcherService != nil || bypassServiceCheck),
+                @"Must have a fetcherService for the test to be able to detect when things finish");
+
+  // Since the callbacks won't be triggered, there isn't a good way to "wait" to confirm the test
+  // has completed unless it triggers the notifications or the test has a fetcher service. If there
+  // is a service, then we can enabled the per host limit, and send a second request to that will
+  // trigger things that can be used for expectations.
+  _fetcherService.maxRunningFetchersPerHost = 1;
+  BOOL useChaser = _fetcherService != nil;
+
+  // If the authorizer/UAProvider/etc is async, then the fetch won't fully begin, and there won't
+  // ever be a start notification (and thus stop notification). Likewise, if the fetch is stopped
+  // before it even starts, there won't be a notification.
+  int expectedStartStopCount = (triggersNotifications ? 1 : 0) + (useChaser ? 1 : 0);
+  int expectedCompletionCount = useChaser ? 1 : 0;
+  FetcherNotificationsCounter *fnctr = [[FetcherNotificationsCounter alloc] init];
+
+  // Use a URL that will timeout, so the fetch takes a while so it can always be cancelled.
+  NSString *timeoutFileURLString = [self localURLStringToTestFileName:kGTMGettysburgFileName
+                                                           parameters:@{@"sleep" : @"5"}];
+  GTMSessionFetcher *fetcher = [self fetcherWithURLString:timeoutFileURLString];
+  // Don't set fetcher.stopFetchingTriggersCompletionHandler
+
+  NSString *fileURLString = [self localURLStringToTestFileName:kGTMGettysburgFileName];
+  GTMSessionFetcher *chaser = (useChaser ? [self fetcherWithURLString:fileURLString] : nil);
+
+  // If notification are expected, use an explicit notification expectation for it to sure
+  // we've waited long enough for that to show up.
+  NS_VALID_UNTIL_END_OF_SCOPE XCTestExpectation *fetcherStoppedNotificationExpectation =
```

Comment:
is the VALID_UNTIL_END_OF_SCOPE required?  I *believe* the test will hang onto expectations created this way because it needs them if you call [self waitForExpectationsWithTimeOut:handler:]